src/org/jitsi/impl/neomedia/transform/rewriting/SsrcGroupRewriter.java

```diff
@@ -414,11 +415,25 @@ private void maybeSwitchActiveRewriter(final RawPacket pkt)
      */
     boolean isKeyFrame(RawPacket pkt)
     {
-        int sourceSSRC = pkt.getSSRC();
-        Byte redPT = ssrcRewritingEngine.ssrc2red.get(sourceSSRC);
-        byte vp8PT = 0x64;
-
-        return Utils.isKeyFrame(pkt, redPT, vp8PT);
+        byte pt = pkt.getPayloadType();
+        Byte redPT = null, codecPT = null;
+        String codecType = null;
+        for (Map.Entry<Byte, MediaFormat> entry :
```

Comment:
@mdaneshi thank you for the updated PR. Could you please have a look at https://github.com/parlaylabs/libjitsi/pull/1 and see if that makes sense? I haven't tested it, but it sketches what I've tried to explain in my previous comment.